### ISSUE TITLE
"USB accessory disabled" problem (ipad pro 11')

### ISSUE THREAD

**Issue by whz97** [tylernguyen/x1c6-hackintosh#65]
Thanks a lot for this wonderful repository and the EFI you have shared! I am using Thinkpad x1c5 5gen 20HR (almost the same as 6gen except the cpu). The Catalina 10.15.5 installation works pretty well after some tweaks, with only one problem.

### Description:
When I connect my ipad pro 11' to this computer via USB, a badge showing "usb accessories disabled unplug the accessory using too much power to re-enable usb devices" pops out and this device cannot be accessed. 

### Reproduce:
Plug my ipad pro 11' to the computer via USB, it is disabled with notification "usb accessories disabled unplug the accessory using too much power to re-enable usb devices". Though this ipad pro is not shown in the sidebar of Finder, it shows in System Report as "ipad" in "USB 3.0 bus":
> Current Available (mA): 500
> Current Required (mA): 500
> Extra Operating Current (mA): 1900
> Sleep Current (mA): 2400

When I connect the same ipad pro to my friend's real Macbook pro, it is recognized normally, and System Report shows:
> Current Available (mA): 500
> Current Required (mA): 500
> Extra Operating Current (mA): 1600
> Sleep Current (mA): 2100

When this ipad pro is connected to the thunderbolt/USB-C port of the Thinkpad, the computer recognizes it normally (after rebooting for the thunderbolt port to work). This problem is not eliminated when the ipad is 100% charged. My iphone connects to the computer via USB with no problem. 

### the EFI I've been using:
see https://github.com/whz97/myEFI/tree/master/EFI.

I am very much a newby to Hackintosh (as you can see), and knows it always has this problem or that. However, the functionality to connect my ipad would really make my daily work a lot easier. I searched for similar issues on google and Hackintosh forums but got nothing very informative, but I suspect it has something to do with the USB power limitations. Could you take a look at this issue and maybe offer some advice?

**Comment by tylernguyen** [comment]
@whz97 

My recommendation would be first upgrading to 10.15.6 since the changelog appears to fix some USB issues.

Secondly, I would play around with com.apple.driver.AppleUSBHostMergeProperties within your USBPort mapping kext. First remove what you have and test, then try lower values until the warning is gone.

**Comment by denghanx** [comment]
@whz97 If your SMIBIOS was the same MacBook Pro 14,1 as me.
You can try add this kext to solve this issue. 

[USBPower.kext.zip](https://github.com/tylernguyen/x1c6-hackintosh/files/4964084/USBPower.kext.zip)

**Comment by tylernguyen** [comment]
https://github.com/tylernguyen/x1c6-hackintosh/issues/34

It seems that I have to play around with usb power injection.

**Comment by whz97** [comment]
I'm terribly sorry for the later follow-up. Thanks @tylernguyen and @denghanx for your kind help. 

Unfortunately, this problem seems more complicated than [#34](https://github.com/tylernguyen/x1c6-hackintosh/issues/34) 

I have tried configuring com.apple.driver.AppleUSBHostMergeProperties within USBPort.kext, and changed the kUSBSleepPortCurrentLimit and kUSBWakePortCurrentLimit to 2100. The problem is only partially solved:

1. The ipad connects to the Hackintosh laptop via USB smoothly. Its data is loaded correctly. Sidecar via USB port is also working normally. 
2. However, the same notification "usb accessories disabled ......." still pops out. In fact, it would pop out and be dismissed for three times until it is completely gone. Moreover, once the laptop sleeps and wakes, this would happen again... The USB connection is not really affected by this.  

I have tried two solutions myself, and neither have worked out:
1. Resetting NVRAM. 
2. Changing kUSBSleepPortCurrentLimit and kUSBWakePortCurrentLimit to lower values, such as 2000, 1500, 1000. Deleting these two entries does not help. The new entries ''kUSBWakePowerSupply'' or ''kUSBSleepPowerSupply'' (found in @denghanx 's USBPort.kext) seem somewhat irrelevant, and adding them has not solved this issue either.

With thunderbolt3 hotplug now fixed, and USB connection effectively working, I can definitely use my ipad with Hackintosh. I don't mean to be a perfectionist, but this notification is obviously annoying. May I enquire about further advice on how to possibly debug this issue?
